src/Validation/Validator.php

```diff
@@ -4,717 +4,799 @@
-   }
-
-   /**
-    * {@inheritDoc}
-    */
-   public function ascii(string $field, ?string $message = null, $when = null)
-   {
-       $message = $message ?? __d('validation', 'ascii');
-
-       return parent::ascii($field, $message, $when);
-   }
-
-   /**
-    * {@inheritDoc}
-    */
-   public function utf8(string $field, ?string $message = null, $when = null)
-   {
-       $message = $message ?? __d('validation', 'utf8');
-
-       return parent::utf8($field, $message, $when);
-   }
-
-   /**
-    * {@inheritDoc}
-    */
-   public function utf8Extended(string $field, ?string $message = null, $when = null)
-   {
-       $message = $message ?? __d('validation', 'utf8Extended');
-
-       return parent::utf8Extended($field, $message, $when);
-   }
-
-   /**
-    * {@inheritDoc}
-    */
-   public function integer(string $field, ?string $message = null, $when = null)
-   {
-       $message = $message ?? __d('validation', 'integer');
-
-       return parent::integer($field, $message, $when);
-   }
-
-   /**
-    * {@inheritDoc}
-    */
-   public function isArray(string $field, ?string $message = null, $when = null)
-   {
-       $message = $message ?? __d('validation', 'isArray');
-
-       return parent::isArray($field, $message, $when);
-   }
-
-   /**
-    * {@inheritDoc}
-    */
-   public function scalar(string $field, ?string $message = null, $when = null)
-   {
-       $message = $message ?? __d('validation', 'scalar');
-
-       return parent::scalar($field, $message, $when);
-   }
-
-   /**
-    * {@inheritDoc}
-    */
-   public function hexColor(string $field, ?string $message = null, $when = null)
-   {
-       $message = $message ?? __d('validation', 'hexColor');
-
-       return parent::hexColor($field, $message, $when);
-   }
-
-   /**
-    * {@inheritDoc}
-    */
-   public function multipleOptions(string $field, array $options = [], ?string $message = null, $when = null)
-   {
-       $message = $message ?? __d('validation', 'multipleOptions');
-
-       return parent::multipleOptions($field, $options, $message, $when);
-   }
-
-   /**
-    * {@inheritDoc}
-    */
-   public function hasAtLeast(string $field, int $count, ?string $message = null, $when = null)
-   {
-       $message = $message ?? __d('validation', 'hasAtLeast {0}', $count);
-
-       return parent::hasAtLeast($field, $count, $message, $when);
-   }
-
-   /**
-    * {@inheritDoc}
-    */
-   public function hasAtMost(string $field, int $count, ?string $message = null, $when = null)
-   {
-       $message = $message ?? __d('validation', 'hasAtMost {0}', $count);
-
-       return parent::hasAtMost($field, $count, $message, $when);
-   }
-
+    /**
+     * @inheritDoc
+     */
+    public function requirePresence(array|string $field, Closure|string|bool $mode = true, ?string $message = null): CakeValidator
+    {
+        $message = $message ?? __d('validation', 'requirePresence');
+
+        return parent::requirePresence($field, $mode, $message);
+    }
```

Comment:
Fix indentation when running composer cs-fix.
Next, follow the argument data type based on the CakePHP5 Validator.
Refer: https://github.com/cakephp/validation/blob/5.x/Validator.php